Signal/src/ViewControllers/Utils/MessageRecipientStatusUtils.swift

```diff
@@ -94,18 +115,32 @@ class MessageRecipientStatusUtils: NSObject {
         }
     }
 
+    // This method is per-message and "biased towards failure".
+    // See comments above.
     public class func statusMessage(outgoingMessage: TSOutgoingMessage,
                                     referenceView: UIView) -> String {
 
+        if outgoingMessage.messageState == .unsent {
```

Comment:
nit: The logical flow of this method seems to mirror`recipientStatus(outgoingMesssage:)`

Can we replace the logic in this method with something like:

```
switch recipientStatus(outgoingMessage: outgoingMessage) {
  case .unsent:
       return NSLocalizedString("MESSAGE_STATUS_FAILED", comment:"message footer for failed messages")
   case .attemptingOut:

...etc...

}

Reply:
Done.